test/services/workflow_executions/preparation_service_test.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module WorkflowExecutions
+  class PreparationServiceTest < ActiveSupport::TestCase
+    def setup
+      @user = users(:john_doe)
+      @workflow_execution = workflow_executions(:irida_next_example)
+    end
+
+    test 'prepare workflow_execution with valid params' do
+      assert_difference -> { ActiveStorage::Attachment.count } => 2 do
+        WorkflowExecutions::PreparationService.new(@workflow_execution, @user, {}).execute
+      end
+
+      assert_equal 1, @workflow_execution.inputs.size
+      assert_equal 1, @workflow_execution.samples_workflow_executions.first.inputs.size
+
+      assert @workflow_execution.workflow_params.key? '--input'
+      assert @workflow_execution.workflow_params.key? '--outdir'
+      assert_equal @workflow_execution.inputs.first.blob.key, @workflow_execution.workflow_params['--input']
+
+      assert_equal 'prepared', @workflow_execution.state
```

Comment:
We should test the state before the preparation is executed as listed in https://github.com/phac-nml/irida-next/issues/246

We don't have a state for it right now but we could just add an assert state not prepared.

Reply:
Fixed in f986ce2d